### ISSUE TITLE
to improve the result

### ISSUE THREAD

**Issue by Usernamezhx** [movchan74/street_to_shop_experiments#3]
hi Movchan 
    thanks your code and blog first. it is so cool. after that do you make any other effort to improve the result ? as you know. the user upload the image with noise. we should focus the main object. add a main part object detect net? or a attention net base the cnn ? I also meet the problem about the domain adaption. I have no good idea. Besides i think the triplet loss based on the cosin or angular will be better

**Comment by mmcctt00** [comment]
I agree with Usernamezhx  Great blog and I suggest you read this paper: 
http://openaccess.thecvf.com/content_ICCV_2017/papers/Wang_Deep_Metric_Learning_ICCV_2017_paper.pdf
Thanks for sharing

**Comment by nok** [comment]
Great post and paper!
